api/room/schema.py

```diff
@@ -38,6 +40,58 @@ def save_room_tags(room, room_tags):
     room.save()
 
 
+class RoomResource(SQLAlchemyObjectType):
+    class Meta:
+        model = RoomResourceModel
+
+
+class AssignResource(graphene.Mutation):
+    """
+        Assigns a resource to a room
+    """
+    class Arguments:
+        resource_id = graphene.Int(required=True)
+        quantity = graphene.Int(required=True)
+        room_id = graphene.Int(required=True)
+    room_resource = graphene.Field(RoomResource)
+
+    @Auth.user_roles('Admin')
+    def mutate(self, info, **kwargs):
+        validate_empty_fields(**kwargs)
+        query = Room.get_query(info)
+        active_room = query.filter(RoomModel.state == "active")
+        exact_room = active_room.filter(
+            RoomModel.id == kwargs['room_id']).first()
+        if not exact_room:
+            raise GraphQLError("Room not found")
+        exact_resource = ResourceModel.query.filter_by(
+            id=kwargs['resource_id']).first()
+        if not exact_resource:
+            raise GraphQLError('invalid resource id')
+        exact_quantity = exact_resource.quantity
+        assigned_quantity = kwargs['quantity']
+        if assigned_quantity > exact_quantity:
+            raise GraphQLError(
+                'assigned resource cannot exceed quantity in database.'
```

Comment:
Nice implementation @rajeman . is it more feasible to automatically update the number of resources when available when a user tries to assign an already assigned resource instead of directing them to a query for `updatingRoomReasource`? 🤔 💭

Reply:
@verenceLola did you mean @d-kahara ?

Reply:
> @verenceLola did you mean @d-kahara ?

actually anyone really.

Reply:
I think it is okay this way. A new story can be created for updating resource

Reply:
As @rajeman says there is a story to update the resource [https://www.pivotaltracker.com/story/show/165014764](here). I think there should be a separation of concerns between updating and assigning a resource.